domains/transfer/Transfer.API/API.IntegrationTests/Claiming/Api/Controllers/ClaimAutomationControllerTest.cs

```diff
@@ -0,0 +1,157 @@
+        result.StatusCode.Should().Be(HttpStatusCode.NoContent);
+    }
+
+    [Fact]
+    public async Task StopProcess_WhenNoClaimProcessExists_ReturnsNotFound()
+    {
+        var subject = Guid.NewGuid();
+
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.DeleteAsync("api/claim-automation/stop");
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+
+    [Fact]
+    public async Task StartProcess_WhenNoClaimProcessHasStarted_ReturnsCreatedAt()
+    {
+        var subject = Guid.NewGuid();
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.PostAsync("api/claim-automation/start", null);
+        result.StatusCode.Should().Be(HttpStatusCode.Created);
+    }
+
+    [Fact]
+    public async Task StartProcess_WhenClaimProcessAlreadyStarted_ReturnsOk()
+    {
+        var subject = Guid.NewGuid();
+
+        await factory.SeedClaims(new List<ClaimSubject>()
+        {
+            new(subject)
+        });
+
+        var client = factory.CreateAuthenticatedClient(sub: subject.ToString());
+
+        var result = await client.PostAsync("api/claim-automation/start", null);
+        result.StatusCode.Should().Be(HttpStatusCode.OK);
+    }
+
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenHistoryEntryHasDifferentSubjectId_ReturnsNotFound()
+    {
+        const string actor = "Peter Producent";
+        var client1 = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString(), actor: actor);
+        await client1.PostAsync("api/claim-automation/start", null);
+
+        var client = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString());
+        var result = await client.GetAsync("api/claim-automation/history");
+
+        result.StatusCode.Should().Be(HttpStatusCode.NotFound);
+    }
+
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenHistoryExists_ReturnsOK()
+    {
+        const string actor = "Peter Producent";
+        var client = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString(), actor: actor);
+        await client.PostAsync("api/claim-automation/start", null);
+        var result = await client
+            .RepeatedlyGetUntil<ClaimSubjectHistoryEntriesDto>(
+                "api/claim-automation/history",
+                res => res.History.Any()
+            );
+        var settings = new VerifySettings();
+        settings.ScrubMember("CreatedAt");
+
+        await Verifier.Verify(result, settings);
+    }
+
+    [Fact]
+    public async Task GetClaimSubjectHistory_WhenMultipleHistoryExists_ReturnsOK()
+    {
+        const string actor = "Peter Producent";
+        var client = factory.CreateAuthenticatedClient(sub: Guid.NewGuid().ToString(), actor: actor);
+        await client.PostAsync("api/claim-automation/start", null);
+        await Task.Delay(5000);
```

Comment:
Why is this delay needed? If it is a work-around, could a different solution be to change the condition in "RepeatedlyGetUntil" to exactly 2 entries instead of "Any"?

Reply:
To be removed